std/format.d

```diff
@@ -583,6 +583,7 @@ can match the expected number of readings or fewer, even zero, if a
 matching failure happens.
  */
 uint formattedRead(R, Char, S...)(ref R r, const(Char)[] fmt, S args)
+    if (!S.length || is(typeof(*args[0])))
```

Comment:
> is(typeof(*args[0]))

Why not `allSatisfy!(isPointer, S)`?

Also, would be clearer to use `S.length == 0`